gateway/models/connections.go

```diff
@@ -586,6 +586,37 @@ func ListConnections(ctx UserContext, opts ConnectionFilterOption) ([]Connection
 	return items, nil
 }
 
+// SearchBySimilarity searches connections by name, type, or subtype using a case-insensitive search.
+func SearchConnectionsBySimilarity(ctx UserContext, searchTerm string) ([]Connection, error) {
```

Comment:
One thing I wanted to raise on this is the `ctx` value we usually pass to the models layer. I have a bit of a hard time finding it necessary, considering that sometimes we need just one or two values from this context and the creation of coupling between the context and the model layer. I would ask @sandromello's opinion as well. From my view, this is unnecessary coupling, and models should get only what they need and nothing else